### PR TITLE
Issue 2756 - Account Selector updates

### LINKED ISSUE #2756: [7] Account Selector should not auto-match on type-ahead version
it's not a good design to auto list some candidates to-account while user type to-account

image

because of this, I have wrongly transfered to account "alpha" while acctually want to transfer to "alpha-go", several times.

image

alt has also mentioned this in forum

https://bitsharestalk.org/index.php?topic=28250.0

please remove this feature.

and maybe one better solution is to set one white list for each wallet, users can only select accounts in white list to send to.

### PR DESCRIPTION
**Pull request by startailcoon**
This PR adds changes to the `<AccountSelector />` component.

To remove the current behavior of automatically select a valid account on input there is a new prop flag `requireActiveSelect`. This flag is default to true, and should probably only be switched to false in edge cases.

This update will require an active select when using the auto complete account selector.

- Add flag to require active select
- Removed duplicate ChainStore calls

Closes #2756

### CODE REVIEW

File: app/components/Modal/SendModal.jsx [hunk 1]
```diff
@@ -244,11 +244,9 @@ class SendModal extends React.Component {
             this.setState(
                 {
                     from_name: np.from_name,
-                    from_account: ChainStore.getAccount(np.from_name),
+                    from_account: np.from_account,
```
Comment (sschiessl-bcp) [inline app/components/Modal/SendModal.jsx]: Why was this done?

Comment (startailcoon) [reply app/components/Modal/SendModal.jsx]: Sorry, I should have documented this change better. 

The component used to look up accounts (`<AccountSelector>`) returns this data from the following query on a change, so no need to query for the same data twice.

https://github.com/bitshares/bitshares-ui/blob/aa9b1f833e1a85c683f2131ec1af826bd06e5db7/app/components/Account/AccountSelector.jsx#L155-L162

It's then set in onFromAccountChange

https://github.com/bitshares/bitshares-ui/blob/aa9b1f833e1a85c683f2131ec1af826bd06e5db7/app/components/Modal/SendModal.jsx#L603-L605

https://github.com/bitshares/bitshares-ui/blob/aa9b1f833e1a85c683f2131ec1af826bd06e5db7/app/components/Modal/SendModal.jsx#L318-L320

**Comment by sschiessl-bcp** [comment]
1. I start typing and select an account. Then i realize its wrong, go back and start typing another name it should clear the selected account in the background when I start typing (not on click).
image

2. Can we adjust the No data logo to appear as if one normal line with small empty box icon left and no account found on the as text? Also, the read "Unknown account" is unnecessary at this point because not visible.
image

**Comment by startailcoon** [comment]
Will supply a fix, good catch

**Comment by startailcoon** [comment]
@sschiessl-bcp updated and ready for merging

**Comment by sschiessl-bcp** [comment]
image

Account loading doesnt seem to work properly. Does it on your end?

**Comment by startailcoon** [comment]
It works for me on my end. Can it be that we need to handle bad nodes not returning anything?

**Comment by startailcoon** [comment]
I believe we need to fix some other things on it because today it seems to request data even if the input hasn't changed or there is a selected account. 

Not sure if this puts any load on the nodes, could be cached(?). But it could be required to refactor this whole component.

Adding a log output every time this is run will output that the component is updated without it being necessary I believe.

https://github.com/bitshares/bitshares-ui/blob/918729f87ad0b9261dd529fb760204184bc42c6a/app/components/Account/AccountSelector.jsx#L311

**Comment by startailcoon** [comment]
This PR will be superseded by another one and is therefor closed.

**Pull request closed without merging**